qiskit/_quantumprogram.py

```diff
@@ -1118,27 +1116,10 @@ def run(self, qobj, wait=5, timeout=60):
         Returns:
             Result: A Result (class).
```

Comment:
This `run()` method in QuantumProgram is still returning a Result, rather than a Job. That's fine in case our goal is backwards compatibility. But the PR doesn't touch the `_wrapper.py` file yet. I think the `execute` function there is the main function that a typical user will use. Should we make that return `Job` instead of the current `Result`? I think so, since an async method is easier to use and plays better with longer executions.

Reply:
Yes I was thinking of leaving QuantumProgram behavior the same but agree with wrapper.execute returning jobs.

Reply:
yeah.. i'm actually removing the `qiskit._compiler.execute()` in the transpiler branch, since it looks like a wrapper around a wrapper. So there would only be `_wrapper.execute()` left, so can you just make that return jobs?

Reply:
actually i think you already did, right? your `_wrapper.execute()` calls `qiskit._compiler.execute()` which returns a job? in that case we are good. I'll just remove that extra indirection in my branch.

Reply:
ok